infra.lvm_tools/shrink_lv/roles/shrink_lv/tasks/main.yaml

```diff
@@ -0,0 +1,52 @@
+---
+- name: Shrink Logical Volumes
+  become: true
+  block:
+   - name: Get kernel version
+     ansible.builtin.set_fact:
+      kernel_version: "{{ ansible_facts.kernel }}"
+   - name: Create a backup of the current initramfs
+     ansible.builtin.copy:
+      remote_src: true
+      src: /boot/initramfs-{{ kernel_version }}.img
+      dest: /boot/initramfs-{{ kernel_version }}.img.{{ backup_extension }}
+      mode: "0600"
+   - name: Copy shrink LV dracut module
+     ansible.builtin.copy:
+      src: "{{ role_path }}/files/"
+      dest: /usr/lib/dracut/modules.d/99shrink_lv
+      mode: "0554"
+   - name: Copy current /etc/fstab to module dir
+     ansible.builtin.copy:
+      remote_src: true
+      src: /etc/fstab
+      dest: /usr/lib/dracut/modules.d/99shrink_lv
+      mode: "0444"
+   - name: Create a new initramfs with the extend boot module
+     ansible.builtin.command: /usr/sbin/dracut -a shrink_lv --kver {{ kernel_version }} --force
+     changed_when: true
+   - name: Reboot the server
+     ansible.builtin.reboot:
+      post_reboot_delay: 30
+   - name: Wait for the host to reboot
```

Comment:
Out of curiosity, why do you need this? Doesn't the reboot command cover it? AFAIK before completing, it connect to the remote machine

Reply:
You're correct, it executes a `whoami` by default, so it should be sufficient.
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/reboot_module.html#parameter-test_command